File: app/frontend/src/types/memory/response.ts
export type MemoryType = "difficulty" | "preference" | "goal" | "progress";

export interface UserMemory {
  id: string;
  type: MemoryType;
  content: string;
  last_seen_at: string;
}

export interface UserMemoryListRead {
  memories: UserMemory[];
}

File: app/backend/schemas/memory/response.py
from datetime import datetime

from pydantic import BaseModel

from app.backend.schemas.memory.models import MemoryType


class UserMemoryRead(BaseModel):
    id: str
    type: MemoryType
    content: str
    last_seen_at: datetime


class UserMemoryListRead(BaseModel):
    memories: list[UserMemoryRead]
